Noggog.CSharpExt/Structs/Numbers/UDouble.cs

```diff
@@ -91,8 +96,9 @@ public int CompareTo(double other)
 
     public static UDouble Parse(string str)
     {
-        TryParse(str, out UDouble ud);
-        return ud;
+        if (TryParse(str, out UDouble ud))
+            return ud;
+        return default;

```

Comment:
Yeah, weird setup I had before.  I almost wonder if it should be throwing.  We have the TryParse alternate, the Parse should throw if it's a failure... or be renamed to ParseOrDefault or something more clear?

Reply:
Parse usually does throw an exception on failure, TryParse true/false. not sure if there is a naming convention for ParseOrDefault (which sounds perfectly understandable to me)